### PR TITLE
Add some keyword

### PR DESCRIPTION
**Pull request by tsandall**
These changes update Rego to include a `var` keyword for explicitly declaring local variables. This addresses the scoping issues we've run into in the past (which has been on the roadmap) for a while.

I've updated a bunch of docs but haven't gone through and tested each example.

### CODE REVIEW

**Comment by tsandall** [comment]
Rebased to appease the linter god.

File: repl/repl.go [hunk 1]
```diff
@@ -951,6 +940,61 @@ func (r *REPL) evalPackage(p *ast.Package) error {
...
+// definition. If the query is a single assignment or equality expression and it
+// can be converted into a rule and compiled, then it will be interpreted as
+// such. This allows users to define constants in the REPL. For example:
+//
+//	> a = 1
+//  > a
+//  1
+//
+// If the expression is an equality statement, then an additional check on the
+// left hand side occurs.opa run For example:
```
Comment (srenatus) [inline repl/repl.go]: Nit: stray "opa run"

**Comment by tsandall** [comment]
Fixed up commits to address comments from @srenatus.

For the time being, it's not required (and the docs have been updated to mention this.) Requiring the keyword would break a lot of existing policies. We should consider requiring it pre-1.0 though. We don't have immediate plans for 1.0 but it's good to keep in mind.

**Comment by srenatus** [comment]
Thanks! I vaguely remember "pragmas" having come up time and again, I
suppose there could be one for that behavior ("strict" comes to mind).

**Comment by tsandall** [comment]
After some discussion with @timothyhinrichs we decided that `some` would be a better name for the keyword than `var`. Also, if/when we introduce a different way of expressing "for-all" we can add a different keyword like `every` which will complement `some` nicely. We arrived at `some`/`every` after surveying a few different languages:

```
some/every    - clojure, javascript
any/all       - python, haskell, elm
exists/forall - f#, ocaml
```

We'll hold off on merging this until the rename has been done.

**Comment by tsandall** [comment]
The keyword has been renamed. I didn't squash the rename changes into respective commits because it would have been a pain to split them up.

**Pull request merged**